### ISSUE TITLE
AnySchema is added to Relations that are defined as ComplexSchemas

### ISSUE THREAD

**Issue by pakrym** [Azure/autorest.modelerfour#255]
AnySchema is added to Relations that are defined as ComplexSchemas

```
 parents: !<!Relations> 
      all:
      - *ref_106
      immediate:
      - *ref_106
```

```
 "Relations": {
      "type": "object",
      "properties": {
        "immediate": {
          "type": "array",
          "items": {
            "$ref": "#/definitions/ComplexSchema"
          }
        },
        "all": {
          "type": "array",
          "items": {
            "$ref": "#/definitions/ComplexSchema"
          }
        }
      },
      "defaultProperties": [],
      "additionalProperties": false,
      "required": [
        "all",
        "immediate"
      ]
    },

```

```
{
  "swagger": "2.0",
  "info": {
    "title": "AutoRest Parameter Flattening",
    "description": "Resource Flattening for AutoRest",
    "version": "1.0.0"
  },
  "host": "localhost:3000",
  "schemes": [
    "http"
  ],
  "produces": [
    "application/json"
  ],
  "consumes": [
    "application/json"
  ],
  "paths": {
    "/originalOperation": {
      "patch": {
        "operationId": "Operation",
        "parameters": [
          {
            "name": "request",
            "in": "query",
            "required": true,
            "type": "string"
          },
          {
            "name": "message",
            "in": "query",
            "required": true,
            "type": "string"
          },
          {
            "name": "scope",
            "in": "query",
            "required": true,
            "type": "string"
          },
          {
            "name": "uri",
            "in": "query",
            "required": true,
            "type": "string"
          },
          {
            "required": true,
            "in": "body",
            "name" : "class",
            "schema": {
              "$ref": "#/definitions/class"
            }
          }
        ],
        "responses": {
          "200": {
            "schema": {
              "$ref": "#/definitions/struct"
            },
            "description": ""
          }
        }
      }
    }
  },
  "definitions": {
    "struct": {
      "type":  "object",
      "allOf": [ { "$ref":  "class" } ],
      "properties": {
        "1": { "type": "string" }
      }
    },
    "class": {
      "type":  "object",
      "properties": {
        "abstract": { "type": "string" },
        "add": { "type": "string" },
        "alias": { "type": "string" },
        "as": { "type": "string" },
        "ascending": { "type": "string" },
        "async": { "type": "string" },
        "await": { "type": "string" },
        "base": { "type": "string" },
        "bool": { "type": "string" },
        "break": { "type": "string" },
        "by": { "type": "string" },
        "byte": { "type": "string" },
        "catch": { "type": "string" },
        "char": { "type": "string" },
        "checked": { "type": "string" },
        "const": { "type": "string" },
        "continue": { "type": "string" },
        "class": { "type": "string" },
        "decimal": { "type": "string" },
        "default": { "type": "string" },
        "delegate": { "type": "string" },
        "descending": { "type": "string" },
        "do": { "type": "string" },
        "double": { "type": "string" },
        "dynamic": { "type": "string" },
        "else": { "type": "string" },
        "enum": { "type": "string" },
        "event": { "type": "string" },
        "explicit": { "type": "string" },
        "extern": { "type": "string" },
        "false": { "type": "string" },
        "finally": { "type": "string" },
        "fixed": { "type": "string" },
        "float": { "type": "string" },
        "for": { "type": "string" },
        "foreach": { "type": "string" },
        "from": { "type": "string" },
        "get": { "type": "string" },
        "global": { "type": "string" },
        "goto": { "type": "string" },
        "group": { "type": "string" },
        "if": { "type": "string" },
        "implicit": { "type": "string" },
        "in": { "type": "string" },
        "int": { "type": "string" },
        "interface": { "type": "string" },
        "internal": { "type": "string" },
        "into": { "type": "string" },
        "is": { "type": "string" },
        "join": { "type": "string" },
        "let": { "type": "string" },
        "lock": { "type": "string" },
        "long": { "type": "string" },
        "nameof": { "type": "string" },
        "namespace": { "type": "string" },
        "new": { "type": "string" },
        "null": { "type": "string" },
        "object": { "type": "string" },
        "on": { "type": "string" },
        "operator": { "type": "string" },
        "orderby": { "type": "string" },
        "out": { "type": "string" },
        "override": { "type": "string" },
        "params": { "type": "string" },
        "partial": { "type": "string" },
        "private": { "type": "string" },
        "protected": { "type": "string" },
        "public": { "type": "string" },
        "readonly": { "type": "string" },
        "ref": { "type": "string" },
        "remove": { "type": "string" },
        "return": { "type": "string" },
        "sbyte": { "type": "string" },
        "sealed": { "type": "string" },
        "select": { "type": "string" },
        "set": { "type": "string" },
        "short": { "type": "string" },
        "sizeof": { "type": "string" },
        "stackalloc": { "type": "string" },
        "static": { "type": "string" },
        "string": { "type": "string" },
        "struct": { "type": "string" },
        "switch": { "type": "string" },
        "this": { "type": "string" },
        "throw": { "type": "string" },
        "true": { "type": "string" },
        "try": { "type": "string" },
        "typeof": { "type": "string" },
        "uint": { "type": "string" },
        "ulong": { "type": "string" },
        "unchecked": { "type": "string" },
        "unmanaged": { "type": "string" },
        "unsafe": { "type": "string" },
        "ushort": { "type": "string" },
        "using": { "type": "string" },
        "value": { "type": "string" },
        "var": { "type": "string" },
        "virtual": { "type": "string" },
        "void": { "type": "string" },
        "volatile": { "type": "string" },
        "when": { "type": "string" },
        "where": { "type": "string" },
        "while": { "type": "string" },
        "yield": { "type": "string" },
        "System": {
          "type": "string",
          "enum": [
            "System",
            "System.Text.Json"
          ],
          "x-ms-enum": {
            "name": "SystemEnum",
            "modelAsString": true
          }
        }
      }
    }
  }
}

```

**Comment by daviwil** [comment]
Hey @pakrym, is there a specific Swagger I can try which exhibits this issue?  It looks like the one fragment for `Relations` above is missing the `ComplexSchema` definition it refers to.

**Comment by daviwil** [comment]
I noticed @joheredi pointed one out in this comment: https://github.com/Azure/autorest.modelerfour/pull/263#issuecomment-615463245.  I'll experiment with that.

**Comment by daviwil** [comment]
@pakrym @joheredi When one of you have a chance, can you try the test build in this PR?

https://github.com/Azure/autorest.modelerfour/pull/305#issuecomment-642871923

**Comment by daviwil** [comment]
Fixed in 4.14.371, verified by @pakrym.  Thanks!
